src/YunInternet/Libvirt/Configuration/Domain/Device/Disk.php

```diff
@@ -168,4 +168,13 @@ public function hasBacking(): bool
     {
         return is_null($this->findChild("backingStore")) === false && $this->backingStore()->isActive();
     }
+
+    /**
+     * @param callable|null $filter A callable accept a BackingStore as parameter
+     * @return array BackingStore[]
+     */
+    public function getBackiStorageCollection($filter = null): array
+    {
+        return $this->getChildren("backingStore", $filter, BackingStore::class, true);
```

Comment:
Do you want to add a default filter here? As you known, there are some situations have an backingStore without type attribute (not active).

Reply:
I don't understand what you mean, can you give me some example of filters?

Reply:
> I don't understand what you mean, can you give me some example of filters?

https://github.com/yzslab/php-libvirt-client/pull/12/files#diff-7481b12aee3e3e747e5273a6c7359bb4e5f41df5082b1c4a037ce042c517ebb9R400 
There are three backingStore element in the XML, but only two of them are active:
```
<backingStore type='...'>
    <backingStore type='...'>
        <backingStore/>
    </backingStore>
</backingStore>
```
I think you should add a default filter here to exclude the not active one.

Reply:
I'm sorry , I have tried, but there was no way, I had never used callable functions

Reply:
> I'm sorry , I have tried, but there was no way, I had never used callable functions

I made some changes, tell me if there are any questions or recommendations.

Reply:
https://github.com/yzslab/php-libvirt-client/pull/12/commits/2bc71a20c96cc718b401416089b5cbf4b5e79200